### PR TITLE
Ocp3 audit config

### LINKED ISSUE #4941: OpenShift API server audit log check formatting errors
#### Description of problem:

The checks for the OpenShift API server audit log configuration have a few issues with regards to the expected config format.  There are 3 checks that are expecting 2 ':' characters after the "auditConfig" string, though only 1 ':' is expected.  In addition the api_server_audit_log_path check is not tolerant of the path being quoted, which causes the check to fail when a quoted patch is provided in the configuration.

#### SCAP Security Guide Version:

master

#### Operating System Version:

NA

#### Steps to Reproduce:

1. Configure OCP with a quoted API server audit log path as shown in the guide example.
2. Run a scan using the ocp3 master profile against a master node.

#### Actual Results:

The api_server_audit_log_path check fails.

#### Expected Results:

The api_server_audit_log_path check should pass.

### PR DESCRIPTION
**Pull request by nkinder**
#### Description:

- This fixes an issue with the expected configuration format for the OpenShift API server audit log checks being incorrect in our regexes.  Specifically, we were expecting an extra ':' character in a few checks, and the audit log file check was not tolerant of quoted paths.

#### Rationale:

- These patches correct the above described issues.

- Fixes #4941

### CODE REVIEW

**Comment by shawndwells** [comment]
@openscap-ci add to whitelist

File: applications/openshift/api-server/api_server_audit_log_path/oval/shared.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
   <ind:textfilecontent54_object id="object_api_server_audit_log_path" version="1">
     <ind:filepath>/etc/origin/master/master-config.yaml</ind:filepath>
-    <ind:pattern operation="pattern match">^[\s]*auditConfig:\:(?:[^\n]*\n+)+?[\s]*auditFilePath\:[\s]+(\S+)[\s]*$</ind:pattern>
+    <ind:pattern operation="pattern match">^[\s]*auditConfig\:(?:[^\n]*\n+)+?[\s]*auditFilePath\:[\s"']+(\S+)[\s"']*$</ind:pattern>
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_audit_log_path/oval/shared.xml]: While an improvement, this regex will match the final ``"``.

For example, if the config stanza is:
`````
auditConfig:
  auditFilePath: "/var/log/audit/ocp.log"
`````

Then the OVAL will match on ``/var/log/audit/ocp.log"``

Played around with the capture group but couldn't figure out how to only parse text within the quotes (if quotes are present).

File: applications/openshift/api-server/api_server_audit_log_maxage/oval/shared.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
   <ind:textfilecontent54_object id="object_api_server_audit_log_maxage" version="1">
     <ind:filepath>/etc/origin/master/master-config.yaml</ind:filepath>
-    <ind:pattern operation="pattern match">^[\s]*auditConfig:(?:[^\n]*\n+)+?[\s]*maximumFileRetentionDays\:[\s]+(\d+)[\s]*$</ind:pattern>
+    <ind:pattern operation="pattern match">^[\s]*auditConfig\:(?:[^\n]*\n+)+?[\s]*maximumFileRetentionDays\:[\s]+(\d+)[\s]*$</ind:pattern>
```
[the remark this answers is not in the document]
Comment (jan-cerny) [reply applications/openshift/api-server/api_server_audit_log_maxage/oval/shared.xml]: OVAL standard says it supports a subset of Perl 5's regular expressions, see here: https://oval.mitre.org/language/about/re_support_5.6.html Effectively, OpenSCAP uses PCRE https://www.pcre.org/ when processing regular expressions.

[the remark this answers is not in the document]
Comment (nkinder) [reply applications/openshift/api-server/api_server_audit_log_maxage/oval/shared.xml]: I can remove the unnecessary escape before the ':' character.  I noticed we used it in already in some of the other checks in this area, so I spied that convention even though it is not a character that requires escaping according to PCRE.

File: applications/openshift/api-server/api_server_audit_log_path/oval/shared.xml (hunk 1 above)
Comment (nkinder) [reply applications/openshift/api-server/api_server_audit_log_path/oval/shared.xml]: We can use a negation in a bracketed character class for the capture group to capture anything that isn't a space or quote character like this snippet:

  auditFilePath\:[\s"']+([^\s"']+)[\s"']*$

I will revise the patch to use this approach.

**Comment by yuumasato** [comment]
LGTM. Thanks for the fixes.
Also changed base to stabilization branch.

**Comment by yuumasato** [comment]
Shawn's requests have been addressed.

**Pull request merged**